checks/file_utils.go

```diff
@@ -131,6 +131,61 @@ func CheckFilesContent(shellPathFnPattern string,
 	return res, nil
 }
 
+// FileCbData is any data the caller can act upon
+// to keep state.
+type FileCbData interface{}
+
+// FileContentCb is the callback.
+// The bool returned indicates whether the CheckFilesContent2
+// should continue iterating over files or not.
+type FileContentCb func(path string, content []byte,
+	dl checker.DetailLogger, data FileCbData) (bool, error)
+
+func CheckFilesContent2(shellPathFnPattern string,
+	caseSensitive bool,
+	c *checker.CheckRequest,
+	onFileContent FileContentCb,
+	data FileCbData,
+) error {
+	predicate := func(filepath string) (bool, error) {
+		// Filter out Scorecard's own test files.
+		if isScorecardTestFile(c.Owner, c.Repo, filepath) {
+			return false, nil
+		}
+		// Filter out files based on path/names using the pattern.
+		b, err := isMatchingPath(shellPathFnPattern, filepath, caseSensitive)
+		if err != nil {
+			return false, err
+		}
+		return b, nil
+	}
+
+	matchedFiles, err := c.RepoClient.ListFiles(predicate)
+	if err != nil {
+		// nolint: wrapcheck
+		return err
+	}
+
+	for _, file := range matchedFiles {
+		content, err := c.RepoClient.GetFileContent(file)
+		if err != nil {
+			//nolint
+			return err
+		}
+
+		ct, err := onFileContent(file, content, c.Dlogger, data)
+		if err != nil {
+			return err
+		}
+
+		if !ct {
+			break
```

Comment:
Why do we break now instead of continuing (as we did before)?

Reply:
depends on what the caller returns. This gives the caller the chance to decide. I think it's more flexible. Resolving the comment. Please re-open if you disagree.

Reply:
I see! In that case it would've been much clearer if `ct` was `continue` :) I thought it stood for "content" or something similar before.

Reply:
i see. continue is a language keyword so I could not use it. let me find a better name then. maybe `cont`?

Reply:
Perhaps `continueIter` or `continueMatch` ?

Reply:
maybe `iterate` or `continueIter`

Reply:
i'll go with `continueIter` since we had the same idea!